### PR TITLE
[SYCL] Use ICD registry keys instead of OCL_ICD_FILENAMES

### PR DESCRIPTION
**Pull request by vladimirlaz**
Change OpenCL CPU and FPGA emulator runtimes configuration on Windows
to use OpenCL ICD registry records instead of OCL_ICD_FILENAMES. That
is done to use the latest OpenCL ICD loader which ignores
OCL_ICD_FILENAMES configuration in the administrative console.

Update documentation accordingly.

### CODE REVIEW

File: sycl/tools/install.bat [hunk 1]
```diff
@@ -12,28 +12,24 @@ IF NOT EXIST %OCL_RT_ENTRY_LIB% (
...
+  set /P "CLEAREXISTING=Should the OCL_ICD_FILENAMES be removed (Y/N): "
+  IF "!CLEAREXISTING!" == "N" (
+    echo Existing configuration is going to be preserved
   ) else (
-    echo WARNING: Incorrect input %EXTENDEXISTING%. Only Y and N are allowed.
-    goto USERINPUT
+    IF "!CLEAREXISTING!" == "Y" (
+      echo Clean up previous configuration
+      REG DELETE "HKLM\SYSTEM\CurrentControlSet\Control\Session Manager\Environment" /f /v OCL_ICD_FILENAMES
+      echo Execute `set OCL_ICD_FILENAMES=` to remove variable from the current concole
```
Comment (romanovvlad) [inline sycl/tools/install.bat:25]:
```suggestion
      echo Execute `set OCL_ICD_FILENAMES=` to remove variable from the current console

```

File: sycl/doc/GetStartedGuide.md [hunk 2]
```diff
@@ -416,10 +416,10 @@ command:
 
     ```bash
     # Install OpenCL FPGA emulation RT
-    # Answer N to clean previous OCL_ICD_FILENAMES configuration
```
Comment (romanovvlad) [inline sycl/doc/GetStartedGuide.md:419]: `install.bat` now requires root privileges, right? Don't we want to support setup thru env var for users who have no such privilege? At least we should mention this requirement.

Comment (oleksandr-pavlyk) [reply sycl/doc/GetStartedGuide.md:419]: Ability to use `OCL_ICD_FILENAMES` is also important in Python packaging, where DPC++ runtimes are installed in user's Python environment, and `OCL_ICD_FILENAMES` is used to help DPC++ runtime find the sycl CPU device.

Comment (vladimirlaz) [reply sycl/doc/GetStartedGuide.md:419]: > `install.bat` now requires root privileges, right? Don't we want to support setup thru env var for users who have no such privilege? At least we should mention this requirement.

The script was always run in the administrative console: https://github.com/intel/llvm/blob/sycl/sycl/doc/GetStartedGuide.md?plain=1#L408

Comment (vladimirlaz) [reply sycl/doc/GetStartedGuide.md:419]: > Ability to use `OCL_ICD_FILENAMES` is also important in Python packaging, where DPC++ runtimes are installed in user's Python environment, and `OCL_ICD_FILENAMES` is used to help DPC++ runtime find the sycl CPU device.

It is not what was introduced by the change. It is reaction to the change in OpenCL RT which ignores OCL_ICD_FILENAMES configuration in the administrative console. For regular usage it is more robust to use registry keys. If you need environment configuration you can still use OCL_ICD_FILENAME in non-administrative console.

Comment (romanovvlad) [reply sycl/doc/GetStartedGuide.md:419]: Ok, thanks for clarification.

**Review by romanovvlad — APPROVED** [review]
LGTM

**Review by bader — APPROVED** [review]

**Pull request merged**